src/Compilers/CSharp/Portable/Symbols/Source/SourceUserDefinedOperatorSymbolBase.cs

```diff
@@ -52,6 +52,29 @@ protected SourceUserDefinedOperatorSymbolBase(
 
             this.CheckUnsafeModifier(declarationModifiers, diagnostics);
 
+            bool isIncrementDecrement = syntax is OperatorDeclarationSyntax { OperatorToken.RawKind: (int)SyntaxKind.PlusPlusToken or (int)SyntaxKind.MinusMinusToken };
+
+            if (isIncrementDecrement)
+            {
+                int parameterCount = ((OperatorDeclarationSyntax)syntax).ParameterList.ParameterCount;
+                if (this.IsStatic)
+                {
+                    if (parameterCount is not (1 or 2))
+                    {
+                        diagnostics.Add(ErrorCode.ERR_BadUnOpArgs, this.GetFirstLocation(), SyntaxFacts.GetText(((OperatorDeclarationSyntax)syntax).OperatorToken.Kind()));
+                    }
+                }
+                else
+                {
+                    Binder.CheckFeatureAvailability(syntax, MessageID.IDS_FeatureUserDefinedCompoundAssignmentOperators, diagnostics, ((OperatorDeclarationSyntax)syntax).OperatorToken.GetLocation());
+
+                    if (parameterCount is not (0 or 2))
```

Comment:
What's the reason for allowing 2 here? I would think that we should prefer giving errors about the specific operator the user typed out, rather than presuming the user meant to type a binary operator instead.

Reply:
> What's the reason for allowing 2 here? I would think that we should prefer giving errors about the specific operator the user typed out, rather than presuming the user meant to type a binary operator instead.

Parser reports an error for the 2 parameters case. I don't want to move all that parameter checking out of the parser, so I just made the smallest change to let "might be a success" cases through in the parser.